pkg/jobmanager/job.go

```diff
@@ -74,28 +89,53 @@ func (j *Job) ExitCode() int {
 	return c
 }
 
-// wait for the underlying process. Broadcast the end via waitChan
-// and close the given resources.
-func (j *Job) wait() {
-	if err := j.cmd.Wait(); err != nil {
-		slog.Debug("Process Wait ended with error", "error", err)
-	}
-
+func (j *Job) Close() {
```

Comment:
Why is this exported? There's no way to use this safely without being `newJob` or `(*Job).wait` (as you need knowledge about `wait` having been called and having returned to avoid the race on `j.cmd.ProcessState`).

Reply:
Updated.